tests/commonlib/kubernetes.py

```diff
@@ -179,7 +187,83 @@ def delete_resources(self, resource_type: str, **kwargs):
+            self.delete_resources(resource_type=resource_type, **kwargs)
+            deleted = self.wait_for_resource(resource_type=resource_type, status_list=['DELETED'], **kwargs)
+
+            if not deleted:
+                raise ValueError(f'could not delete Pod: {kwargs}')
+
+            pod = self.create_patched_resource(resource_type, patch_body)
+
+            return pod
+
+        else:
+            return self.dispatch_patch[resource_type](**kwargs)
+    
+    def create_patched_resource(self, patch_resource_type, patch_body):
+        file_path = Path(__file__).parent / '../deploy/mock-pod.yml'
+        k8s_resources = get_k8s_yaml_objects(file_path=file_path)
+
+        patch_metadata = patch_body['metadata']
+        patch_relevant_metadata = {k: patch_metadata[k] for k in ('name', 'namespace') if k in patch_metadata}
+
+        patched_resource = None
+
+        for yml_resource in k8s_resources:
+            resource_type, metadata = yml_resource['kind'], yml_resource['metadata']
+            relevant_metadata = {k: metadata[k] for k in ('name', 'namespace') if k in metadata}
+
+            if resource_type != patch_resource_type or relevant_metadata != patch_relevant_metadata:
+                continue
+            
+            patched_body = self.patch_resource_body(yml_resource, patch_body)
+            created_resource = self.create_from_dict(patched_body, **relevant_metadata)
+
+            done = self.wait_for_resource(resource_type=resource_type, status_list=["RUNNING", "ADDED"], **relevant_metadata)
+            if done:
+                patched_resource = created_resource
+
+            break
+        
+        return patched_resource
+    
+    def patch_resource_body(self, body: Union[list,dict], patch: Union[list,dict]) -> Union[list,dict]:
+        """
+        """
+        if type(body) != type(patch):
+            raise ValueError(f'Cannot compare {type(body)}: {body} with {type(patch)}: {patch}')
+        
+        if isinstance(body, dict):
+            for key, val in patch.items():
+                if key not in body:
+                    body[key] = val
+                else:
+                    if isinstance(val, list) or isinstance(val, dict):
+                        body[key] = self.patch_resource_body(body[key], val)
+                    else:
+                        body[key] = val
+        
+        elif isinstance(body, list):
+            for i, val in enumerate(body):
+                if i >= len(patch):
+                    break
+
+                if isinstance(val, list) or isinstance(val, dict):
+                    body[i] = self.patch_resource_body(body[i], val)
+                else:
+                    body[i] = val
+            
+            if len(patch) > len(body):
+                body += val[len(patch):]
+                
+        else:
+            raise ValueError(f'Invalid body {body} of type {type(body)}')
+        
+        return body
 
```

Comment:
Very complicated function that includes many if else statements and recursion.
Is it possible to improve?

Reply:
Discussed that this can be refactored at a later time @gurevichdmitry

Reply:
new [task](https://github.com/elastic/security-team/issues/4295) is created